src/mvba/abba/mod.rs

```diff
@@ -21,11 +22,10 @@ pub(crate) const MODULE_NAME: &str = "abba";
 
 /// The ABBA holds the information for Asynchronous Binary Byzantine Agreement protocol.
 pub(crate) struct Abba {
-    id: String,                  // this is same as $ID$ in spec
-    i: NodeId,                   // this is same as $i$ in spec
-    r: usize,                    // this is same as $r$ in spec
+    tag: Tag,                    // this takes the place of $ID$ in spec
```

Comment:
Tag has information about round and initiator id. Keeping `round` and `id` next to tag is kind of redundant information.

Reply:
I think round is different from the Tag sequence number. The tag sequence number tells you which instance of MVBA we are running, i.e. it could be the 23rd MVBA decision and round tells us which round of ABBA we are in.

I think we need to keep those distinct.

The `j` in Tag and the `id` here is confusing me. I thought `id` is our own node id. the J in tag is the proposer.

Am I wrong here?

Reply:
> ID is used to distinctness different instance of MVBA. In this case sequence number is always zero for all instances of VCBC.

I was thinking of it as the `ID + seq` number is what distinguishes different instances of MVBA.

i.e. say we are using MVBA for membership. The ID could be "safe-network-membership" and then the sequence number would be incremented for every membership change. e.g. Join(Node A) would have a tag `("safe-network-membership", 0)`, then `Leave(Node B)` would have tag `("safe-network-membership", 1)`, etc..

The thinking behind this sequence number is to allow for gossip protocols to share the "chain" of decisions. i.e. one node has seen all decisions up to 34, another node has seen 39. The second node knows they are ahead and can send the newer decisions to the first node.

So modifying the combinations you listed:
```
MVBA protocols distinguishes by ID + seq
ABBA protocols distinguishes by ID + seq + round + initiator Node ID
VCBC protocols distinguishes by ID + seq + initiator Node ID
```

Reply:
We can again call (ID + seq) the consensus ID. SO internally we don't have a sequence number.  SO the application might have a sequence number.